python/hail/methods/io.py

```diff
@@ -0,0 +1,116 @@
+from hail.typecheck import *
+from hail.utils.java import Env, handle_py4j, joption
+from hail.api2 import MatrixTable
+from hail.history import *
+from hail.expr.types import *
+from .misc import require_biallelic
+
+@handle_py4j
+### @write_history('output', parallel='parallel')
+@typecheck(dataset=MatrixTable,
+           output=strlike,
+           append_to_header=nullable(strlike),
+           parallel=nullable(enumeration('separate_header', 'header_per_shard')),
+           metadata=nullable(dictof(strlike, dictof(strlike, dictof(strlike, strlike)))))
+def export_vcf(dataset, output, append_to_header=None, parallel=None, metadata=None):
+    """Export variant dataset as a .vcf or .vcf.bgz file.
+
+    .. include:: ../_templates/req_tvariant.rst
+
+    Examples
+    --------
+    Export to VCF as a block-compressed file:
+
+    >>> methods.export_vcf(dataset, 'output/example.vcf.bgz')
+
+    Notes
+    -----
+    :meth:`export_vcf` writes the VDS to disk in VCF format as described in the
+    `VCF 4.2 spec <https://samtools.github.io/hts-specs/VCFv4.2.pdf>`__.
+
+    Use the ``.vcf.bgz`` extension rather than ``.vcf`` in the output file name
+    for `blocked GZIP <http://www.htslib.org/doc/tabix.html>`__ compression.
+
+    Note
+    ----
+        We strongly recommended compressed (``.bgz`` extension) and parallel
+        output (``parallel=True``) when exporting large VCFs.
+
+    Hail exports the fields of Struct `info` as INFO fields,
+    the elements of Set[String] `filters` as FILTERS, and the
+    value of Float64 `qual` as QUAL. No other row fields are exported.
+
+    The FORMAT field is generated from the genotype schema, which
+    must be a :py:class:`~hail.expr.TStruct`.  There is a FORMAT
+    field for each field of the Struct.
+
+    INFO and FORMAT fields may be generated from Struct fields of type Call,
+    Int32, Float32, Float64, or String. If a field has type Int64, every value
+    must be a valid Int32. Arrays and Sets containing these types are also
+    allowed but cannot be nested; for example, Array[Array[Int32]] is invalid.
+    Sets and Arrays are output with the same comma-separated format. Boolean
```

Comment:
are output => are written

Reply:
do we really support sets?

Reply:
Why is Boolean listed separately from the other supported types?

Reply:
Yes, we support sets and export them just as we do arrays. Boolean is listed separately from the other supported types because Flags "can only appear in INFO fields" according to the VCF Spec 4.2.